### PR TITLE
Update on-boarding workflow and add tutorial carousel

### LINKED ISSUE #147: Create new on-boarding screens
When implemented this feature will allow the application to show a carousel of on-boarding training screens. They will only be displayed once and their content will be customizable. See [these mock designs](https://xd.adobe.com/view/93c2d0e1-081a-4f62-b358-4e3d9b9a8847-c691/grid/) to better understand how it will be presented when done.

### PR DESCRIPTION
**Pull request by jleach**
# Summary of Changes

This PR implements the on-boarding workflow which includes:

- A Splash screen to load state (Settings, config, etc);
- A Tutorial with carousel pages meant to convey important information or instructions;
- Persistent state manage from a a central location.

The Splash screen is moved from native iOS and Android to be a RN screen where state or configuration can be loaded. The splash screen then routes users as follows: If it's their first time using the app they are shown the Terms and Conditions followed by the Tutorial carousel and finally the pin creation. If they have used the application before they are taken to the pin entry screen.

The persistatne state is the found ate to other state, like settings, that we'll want backed by persistent storage and accessible throughout the app. By using the State + Reducer pattern it can be updated in one location (settings) and immediately impact all areas of the app that depend on this state.

One additional change is moving theming to be more typescript friendly, for example, Colors can be imported with `import { Colors } from 'Theme'` rather than individually and TypeScript can be leverage for auto completion to determine what colours are available. Other components can be added here to better for example the basics for a button (how wide is the boarder and does it have rounded corners or not) for a more robust theming experience.

Onboarding

To reset on-boarding add the like:

```JavaScript
await AsyncStorage.removeItem(LocalStorageKeys.Onboarding)
```

To the `init` function in `Spash.tsx` (~ line 51).

# Related Issues

Fixes #147

# Pull Request Checklist

This is just a reminder about the most common mistakes. Please make sure that you tick all _appropriate_ boxes. But please read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!

If an item doesn't apply to your pull request, **check it anyway** to make it apparent that there's nothing to do.

- [x] All commits contain a DCO `Signed-off-by` line (we use the [DCO GitHub app](https://github.com/apps/dco) to enforce this).
- [x] Updated LICENSE-3RD-PARTY.md for any added dependencies or vendored components.
- [x] Run prettier: `npm run style-format`
- [x] Updated **documentation** for changed code and new or modified features.

If you have _any_ questions to _any_ of the points above, just **submit and ask**! This checklist is here to _help_ you, not to deter you from contributing!

_PR template adapted from the Python attrs project._

### CODE REVIEW

**Comment by jleach** [comment]
@amanji @NeilSMyers What do you two think of the direction of this PR? I think this will end up as a `npm` package to make it as easy as possible for implementers to create their own solution. This is an example of how it would be implemented by anyone:

```
      <Stack.Screen name="Contacts">
        {(props) => (
          <Onboarding {...props}
            title={title} 
            pages={pages} 
            onOnboardingDismissed={markTutorialFin} 
            style={styles} />
        )}
      </Stack.Screen>
```

**Comment by amanji** [comment]
I think it's a good implementation. Later there's the option of enhancing it to pass pages directly as child components.

File: App/navigators/AuthenticateStack.tsx [hunk 1]
```diff
@@ -49,14 +47,13 @@ const AuthenticateStack: React.FC<Props> = ({ setAuthenticated }) => {
   return (
     <Stack.Navigator screenOptions={{ ...defaultStackOptions, presentation: 'transparentModal', headerShown: false }}>
       {firstLogin ? (
-        <Stack.Group>
-          <Stack.Screen name="Terms & Conditions" component={Terms} />
-          <Stack.Screen name="Create 6-Digit Pin" component={PinCreate} initialParams={{ setAuthenticated }} />
-        </Stack.Group>
+        // <Stack.Group>
```
Comment (amanji) [inline App/navigators/AuthenticateStack.tsx]: We can just remove the groups if they are single screens now

File: App/screens/Scan.tsx [hunk 2]
```diff
@@ -44,23 +44,33 @@ const Scan: React.FC<Props> = ({ navigation }) => {
       type: 'info',
       text1: t('Scan.AcceptingConnection'),
     })
+    let msg = 0
```
Comment (amanji) [inline App/screens/Scan.tsx]: Can these be removed?

File: App.tsx [hunk 3]
```diff
@@ -57,23 +58,15 @@ const App = () => {
...
-    return <Splash />
-  }
+  }, [])
 
   return (
-    <AgentProvider agent={agent}>
-      <RootStack />
-      <Toast topOffset={15} config={toastConfig} />
-    </AgentProvider>
+    <Store>
```
Comment (amanji) [inline App.tsx]: Small nitpick: for consistency, can we call this `StoreProvider`?

**Review by amanji — CHANGES_REQUESTED** [review]

File: App/navigators/AuthenticateStack.tsx (hunk 1 above)
Comment (jleach) [reply App/navigators/AuthenticateStack.tsx]: Fixed

File: App/screens/Scan.tsx (hunk 2 above)
Comment (jleach) [reply App/screens/Scan.tsx]: Fixed.

File: App.tsx (hunk 3 above)
Comment (jleach) [reply App.tsx]: Fixed.

File: App/navigators/RootStack.tsx [hunk 4]
```diff
@@ -1,36 +1,96 @@
...
-          </Stack.Screen>
-        </Stack.Group>
-      ) : (
-        <Stack.Group>
-          <Stack.Screen name="Authenticate" options={{ presentation: 'modal' }}>
-            {() => <AuthenticateStack setAuthenticated={setAuthenticated} />}
-          </Stack.Screen>
-        </Stack.Group>
+    <Stack.Navigator initialRouteName={Screens.Splash} screenOptions={{ ...defaultStackOptions, headerShown: false }}>
+      {!authenticated && (
```
Comment (amanji) [inline App/navigators/RootStack.tsx]: Should we still be showing the Tabs and Connect screens if the user is not authenticated?

**Review by amanji — COMMENTED** [review]
Only thing I'd say is try to remove any commented out code if it is un-used. I've also generally followed the repo and used `Props` as the component props interface name. If we want to be more specific about naming props interfaces, we should probably align the other screens with that.

**Comment by jleach** [comment]
> Only thing I'd say is try to remove any commented out code if it is un-used. 

Removed the dead code. It was actually already gone I was just using to the wrong branch. 

> I've also generally followed the repo and used `Props` as the component props interface name. If we want to be more specific about naming props interfaces, we should probably align the other screens with that.

I'll align them with custom names. I wasn't sure if we over use `Props` if VSCode / TypeScript becomes less useful. Rather than say something is wrong with `Props` I'd rather see something is wrong with `AuthStackProps`. Not 100% its an issue.

**Comment by amanji** [comment]
Yeah thinking it's better to be more specific than have numerous `Props` results show up when searching.

**Review by amanji — APPROVED** [review]

**Pull request merged**